src/components/slide/selection-frame.tsx

```diff
@@ -221,48 +205,83 @@ export const SelectionFrame: React.FC<Props> = ({ children, treeId }) => {
 
   const isHovered = hoveredElementId === children.props.id;
   const isSelected = editableElementId === children.props.id;
-  const isSelectedAndMarkdown = isSelected && isMdElement(selectedElement);
+
+  const handleMouseDown = useCallback(
+    (event: MouseEvent<HTMLDivElement>) => {
+      if (
+        (event.target as HTMLElement).classList.contains('moveable-control')
+      ) {
+        return;
+      }
+      event.stopPropagation();
+      dispatch(deckSlice.actions.editableElementSelected(children.props.id));
+
+      if (event.detail === 2) {
+        setDoubleClickedElement(true);
+      }
+    },
+    [children.props.id, dispatch]
+  );
+
+  useEffect(() => {
+    if (doubleClickedElement && selectedElement?.id !== children.props.id) {
+      setDoubleClickedElement(false);
+    }
+  }, [children.props.id, doubleClickedElement, selectedElement?.id]);
+
+  const isEditingMarkdown =
+    doubleClickedElement && isMdElement(selectedElement);
+
+  const editingFrameMetrics = children.props.componentProps?.isFreeMovement
+    ? {
+        left: children?.props?.componentProps?.positionX || 0,
+        top: children?.props?.componentProps?.positionY || 0,
+        width: children?.props?.width,
```

Comment:
Seems like most of the `?`s are unnecessary here.

Reply:
<img width="588" alt="Screen Shot 2022-08-23 at 9 55 47 AM" src="https://user-images.githubusercontent.com/1738349/186191227-471a767b-2b3f-4e7b-90e0-8efaefccebe1.png">

Agreed. Right now children elements don't have a type and that's something to add to ensure the right stuff gets passed as children to the frame.

Reply:
Types on children are always tough 😓 In this spot, at least you can make the assumption that you have objects at least up until `children.props.componentProps` because you have the condition on the ternary guaranteeing that. 

Maybe as long as the children present an unknown, you could gather all the child-prop-grabbing logic into one spot in the component code. If you do a check that makes you confident in the child identity, you could make a type assertion on the props `(children.props as MyExpectedChildComponentProps)`, and gather all your uncertainty into one spot.